### ISSUE TITLE
Unable to set label on node

### ISSUE THREAD

**Issue by jsnod** [jenkinsci/ec2-fleet-plugin#254]
Since upgrading from v2.2.2 to v2.3.1 Jenkins Server is unable to use executors. The instances spin up just fine in AWS, but Jenkins fails with the following:

```FleetSingleExec [single_executor] start
Jan 20, 2021 5:37:27 PM INFO com.amazon.jenkins.ec2fleet.NoDelayProvisionStrategy apply
label [multiple_executor]: currentDemand 0 availableCapacity 4 (availableExecutors 0 connectingExecutors 0 plannedCapacitySnapshot 4 additionalPlannedCapacity 0)
Jan 20, 2021 5:37:27 PM INFO com.amazon.jenkins.ec2fleet.NoDelayProvisionStrategy apply
label [multiple_executor]: currentDemand is less than 1, not provisioning
Jan 20, 2021 5:37:27 PM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
FleetSingleExec [single_executor] fleet instances: []
Jan 20, 2021 5:37:27 PM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
FleetSingleExec [single_executor] described instances: []
Jan 20, 2021 5:37:27 PM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
FleetSingleExec [single_executor] jenkins nodes: []
Jan 20, 2021 5:37:37 PM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
FleetMultipleExec [multiple_executor] start
Jan 20, 2021 5:37:37 PM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
FleetMultipleExec [multiple_executor] fleet instances: [i-09f45995162f9da9b]
Jan 20, 2021 5:37:37 PM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
FleetMultipleExec [multiple_executor] described instances: [i-09f45995162f9da9b]
Jan 20, 2021 5:37:37 PM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
FleetMultipleExec [multiple_executor] jenkins nodes: []
Jan 20, 2021 5:37:37 PM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
FleetMultipleExec [multiple_executor] new instance(s): [i-09f45995162f9da9b] not yet registered as nodes in Jenkins
Jan 20, 2021 5:37:37 PM WARNING com.amazon.jenkins.ec2fleet.EC2FleetCloud warning
FleetMultipleExec [multiple_executor] Unable to set label on node
java.lang.NullPointerException
	at com.amazon.jenkins.ec2fleet.EC2FleetCloud.addNewSlave(EC2FleetCloud.java:783)
	at com.amazon.jenkins.ec2fleet.EC2FleetCloud.access$100(EC2FleetCloud.java:57)
	at com.amazon.jenkins.ec2fleet.EC2FleetCloud$2.run(EC2FleetCloud.java:662)
	at hudson.model.Queue._withLock(Queue.java:1398)
	at hudson.model.Queue.withLock(Queue.java:1275)
	at com.amazon.jenkins.ec2fleet.EC2FleetCloud.updateByState(EC2FleetCloud.java:657)
	at com.amazon.jenkins.ec2fleet.EC2FleetCloud.update(EC2FleetCloud.java:490)
	at com.amazon.jenkins.ec2fleet.CloudNanny.doRun(CloudNanny.java:58)
	at hudson.triggers.SafeTimerTask.run(SafeTimerTask.java:91)
	at jenkins.security.ImpersonatingScheduledExecutorService$1.run(ImpersonatingScheduledExecutorService.java:58)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

Jan 20, 2021 5:37:37 PM INFO com.amazon.jenkins.ec2fleet.EC2FleetCloud info
```

I've had to downgrade to v2.2.2 in order to get any useful executor agents.

**Comment by haugenj** [comment]
Thanks for the logs they were very helpful!

This is caused by the new instance variable we introduced, `maxTotalUses`, not getting initialized. It's only initialized in the constructor of EC2FleetCloud, which is only called when the plugin config is changed. 

You should be able to use version 2.3.1 if you want that feature right now - after updating go to the configuration page and press save. This will update the fleets in config.xml and prevent the null pointer exception for the new variable. 

Otherwise I've merged a fix for this bug, and will try to release it later today or tomorrow. That should make it safe to update the plugin without having to touch the config page.

**Comment by haugenj** [comment]
Released in 2.3.2
